### PR TITLE
Fixed issue 116

### PR DESCRIPTION
**Pull request by Barre14**
I just added the if condition to check whether arrayToken != null. This seemed to work.

### CODE REVIEW

**Comment by Courela** [comment]
You're right, arrayToken may be null if no node if found. You could do this more easily just appending a ? in front of arrayToken?.Type != JTokenType...

File: JUST.net/JsonTransformer.cs [hunk 1]
```diff
@@ -289,61 +289,64 @@ private void LoopOperation(JProperty property, JArray parentArray, JToken curren
...
 
-            if (arrayToken is IDictionary<string, JToken> dict) //JObject is a dictionary
+            if (arrayToken != null)
             {
-                isDictionary = true;
-                JArray arr = new JArray();
-                foreach (var item in dict)
+
+                //workaround: result should be an array if path ends up with array filter
+                if (typeof(T) == typeof(JsonPathSelectable) && Regex.IsMatch(strArrayToken ?? string.Empty, "\\[.+\\]$") && arrayToken.Type != JTokenType.Array)
```
Comment (Courela) [inline JUST.net/JsonTransformer.cs:297]: ...here

**Review by Courela — APPROVED** [review]

**Comment by Barre14** [comment]
ok, thx. Your suggestion would idd work too :).

thx for approving the pull request. Do you know who can make a new build of the package?

**Comment by Courela** [comment]
Only @WorkMaze

**Pull request merged**